src/Cli/dotnet/Commands/Test/Terminal/TerminalTestReporter.cs

```diff
@@ -168,6 +168,11 @@ public void AssemblyRunStarted(string assembly, string? targetFramework, string?
         var assemblyRun = GetOrAddAssemblyRun(assembly, targetFramework, architecture, executionId);
         assemblyRun.Tries.Add(instanceId);
 
+        // If we fail to parse out the parameter correctly this will enable retry on re-run of the assembly within the same execution.
+        // Not good enough for general use, because we want to show (try 1) even on the first try, but this will at
+        // least show (try 2) etc. So user is still aware there is retry going on, and counts of tests won't break.
+        _isRetry |= assemblyRun.Tries.Count > 1;
```

Comment:
nit:
I was briefly confused why you were using (strictly) greater than 1 and suspected an off-by-one error, but it was just because you put this after adding instanceId to assemblyRun.Tries, so I think this is actually correct.

Then I wondered if this had to be in this order because assemblyRun.Tries is a HashSet, and Add might not add it, but it's just a List.

Do you think a HashSet makes more sense? Or put another way, is it expected that you would sometimes get the same instanceId here twice? If not, can we move this before the Add call and make it > 0?

Reply:
We don't expect getting two of the same instance ids, if that happens it is a bug, and hashset will make that harder to debug.

To me the code makes sense as is. We are doing more than 1 try, which is a retry, so we enable retry.

Reply:
> apart from the current code already having a green build without conflicts :)))

I find this a very convincing argument 😄